doc/reference/autoinstall-reference.rst

```diff
@@ -3,184 +3,1013 @@
+
+The channel to check for updates.
 
 .. _ai-keyboard:
 
 keyboard
-========
+--------
 
-stub
+* **type:** mapping, see below
+* **default:** US English keyboard
+* **can be interactive:** yes
 
-.. _ai-late-commands:
+The layout of any attached keyboard. Often systems being automatically
+installed will not have a keyboard at all in which case the value used here
+does not matter.
 
-late-commands
-=============
+The mapping's keys correspond to settings in the :file:`/etc/default/keyboard`
+configuration file. See
+`its manual page <http://manpages.ubuntu.com/manpages/bionic/en/man5/keyboard.5.html>`_
+for more details.
 
-stub
+The mapping contains keys:
 
-.. _ai-locale:
+layout
+~~~~~~
 
-locale
-======
+* **type:** string
+* **default:** ``"us"``
+
+Corresponds to the ``XKBLAYOUT`` setting.
+
+variant
+~~~~~~~
+
+* **type:** string
+* **default:** ``""``
+
+Corresponds to the ``XKBVARIANT`` setting.
+
+toggle
+~~~~~~
+
+* **type:** string or null
+* **default:** ``null``
 
-stub
+Corresponds to the value of ``grp:`` option from the ``XKBOPTIONS`` setting.
+Acceptable values are (but note that the installer does not validate these):
+``caps_toggle``, ``toggle``, ``rctrl_toggle``, ``rshift_toggle``,
+``rwin_toggle``, ``menu_toggle``, ``alt_shift_toggle``, ``ctrl_shift_toggle``,
+``ctrl_alt_toggle``, ``alt_caps_toggle``, ``lctrl_lshift_toggle``,
+``lalt_toggle``, ``lctrl_toggle``, ``lshift_toggle``, ``lwin_toggle``,
+``sclk_toggle``
+
+The version of Subiquity released with 20.04 GA does not accept ``null`` for
+this field due to a bug.
+
+.. _ai-source:
+
+source
+------
+
+* **type:** mapping, see below
+* **default:** see below
+* **can be interactive:** yes
+
+search_drivers
+~~~~~~~~~~~~~~
+
+* **type:** boolean
+* **default:** ``true``
+
+Whether the installer should search for available third-party drivers. When
+set to ``false``, it disables the drivers :ref:`screen and section<ai-drivers>`.
+
+id
+~~
+
+* **type:** string
+* **default:** identifier of the first available source.
+
+Identifier of the source to install (e.g., ``"ubuntu-server-minimal"``).
 
 .. _ai-network:
 
 network
-=======
+-------
 
-stub
+* **type:** netplan-format mapping, see below
+* **default:** DHCP on interfaces named ``eth*`` or ``en*``
+* **can be interactive:** yes
 
-.. _ai-oem:
+`Netplan-formatted <https://netplan.io/reference>`_ network configuration.
+This will be applied during installation as well as in the installed system.
+The default is to interpret the config for the install media, which runs
+DHCPv4 on any interface with a name matching "``eth*``" or "``en*``" but then
+disables any interface that does not receive an address.
 
-oem
-===
+For example, to run DHCPv6 on a particular NIC:
 
-stub
+.. code-block:: yaml
 
-.. _ai-packages:
+    network:
+      version: 2
+      ethernets:
+        enp0s31f6:
+          dhcp6: true
 
-packages
-========
+Note that because of a bug, the version of Subiquity released with 20.04 GA
```

Comment:
Should we link to the bug here?

Reply:
I found the commit in question, but not a bug number.  I'm not sure this has a specific bug number, maybe we focus less on the bug part and just say "20.04 GA behavior is ..."

Reply:
Long term I want to do the versioned docs thing so someone can pull up subiquity version X and see a version of the autoinstall reference correct for that release.  That may be difficult for old releases but it would be nice to make new ones look like that.

Reply:
confirmed https://github.com/canonical/subiquity/pull/739 - the discourse report is good but doesn't offer much additional context, I suggest omitting a link.

Reply:
Understood. Thanks for clarifying!